astropy/table/table.py

```diff
@@ -785,35 +785,55 @@ def _new_from_slice(self, slice_):
         table.meta.clear()
         table.meta.update(deepcopy(self.meta))
         table.primary_key = self.primary_key
-        cols = self.columns.values()
 
         newcols = []
-        for col in cols:
-            col.info._copy_indices = self._copy_indices
+        for col in self.columns.values():
             newcol = col[slice_]
-            if col.info.indices:
+
+            # Note in line below, use direct attribute access to col.indices for Column
+            # instances instead of the generic col.info.indices.  This saves about 4 usec
+            # per column.
+            if (col if isinstance(col, Column) else col.info).indices:
+                # TODO : as far as I can tell the only purpose of setting _copy_indices
+                # here is to communicate that to the initial test in `slice_indices`.
+                # Why isn't that just sent as an arg to the function?
+                col.info._copy_indices = self._copy_indices
                 newcol = col.info.slice_indices(newcol, slice_, len(col))
+
+                # Don't understand why this is forcing a value on the original column.
+                # Normally col.info does not even have a _copy_indices attribute.  Tests
+                # still pass if this line is deleted.  (Each col.info attribute access
+                # is expensive).
+                col.info._copy_indices = True
+
             newcols.append(newcol)
-            col.info._copy_indices = True
 
-        self._make_table_from_cols(table, newcols)
+        self._make_table_from_cols(table, newcols, check_cols=False,
+                                   names=self.columns.keys())
         return table
 
     @staticmethod
-    def _make_table_from_cols(table, cols):
+    def _make_table_from_cols(table, cols, check_cols=True, names=None):
         """
         Make ``table`` in-place so that it represents the given list of ``cols``.
         """
-        colnames = set(col.info.name for col in cols)
-        if None in colnames:
-            raise TypeError('Cannot have None for column name')
-        if len(colnames) != len(cols):
-            raise ValueError('Duplicate column names')
+        if check_cols:
+            colnames = set(col.info.name for col in cols)
+            if None in colnames:
+                raise TypeError('Cannot have None for column name')
+            if len(colnames) != len(cols):
+                raise ValueError('Duplicate column names')
+
+        if names is None:
+            names = (col.info.name for col in cols)
 
-        columns = table.TableColumns((col.info.name, col) for col in cols)
+        columns = table.TableColumns((name, col) for name, col in zip(names, cols))
 
         for col in cols:
-            col.info.parent_table = table
+            # For Column instances it is much faster to do direct attribute access
+            # instead of going through .info
```

Comment:
Again, perhaps add a `TODO` to remind us to look at `info` more carefully...

Reply:
This is the second item on the Table performance Wiki and it is not going to escape attention.  Basically as soon as you start profiling this comes up.  So probably no need to clutter the code with TODO for this.

Reply:
OK, makes sense to just with the wiki